src/context.cpp

```diff
@@ -483,6 +484,102 @@ namespace librealsense
         query_option_callback _query_opt_cb;
     };
 
+
+    // Frame data and metadata are sent as two seperate streams.
+    // They are synchronized using frame_id for matching data+metadata sets.
+    template< class T >
```

Comment:
Why a template?

Reply:
Motion frame type is different from Video frame type.
I did not want to use a general pointer to support move operations.